pkg/controller/master/migration/vmim_controller.go

```diff
@@ -88,3 +92,176 @@ func (h *Handler) syncVM(vmi *kubevirtv1.VirtualMachineInstance) error {
 	_, err = h.vms.Update(toUpdateVM)
 	return err
 }
+
+// scaleResourceQuota scales the resource quota of the namespace to allow the migration to succeed
+func (h *Handler) scaleResourceQuota(vmi *kubevirtv1.VirtualMachineInstance) error {
+	// If the namespace is not managed by the resource quota, skip scaling
+	if exist, err := h.isNamespaceManagedByResourceQuota(vmi.Namespace); exist == false && err == nil {
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("failed to check if the namespace is managed by the resource quota: %v", err)
+	}
+
+	// Scale ResourceQuota through VMI resource specifications
+	if err := h.scaleResourceQuotaWithVMI(vmi); err != nil {
+		return fmt.Errorf("failed to scale resource quota with vmi: %v", err)
+	}
+
+	return nil
+}
+
+// isNamespaceManagedByResourceQuota checks if the namespace is managed by the resource quota
+func (h *Handler) isNamespaceManagedByResourceQuota(namespace string) (bool, error) {
+	rqs, err := h.rqCache.List(namespace, labels.Everything())
+	if err != nil {
+		return false, err
+	}
+
+	// If there is any resource quota in the namespace,
+	// return true check it if finding cpu or memory limit.
+	for _, v := range rqs {
+		if v.Spec.Hard.Cpu() != nil || v.Spec.Hard.Memory() != nil {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// scaleResourceQuotaWithVMI Scaling ResourceQuota through VMI resource specifications
+func (h *Handler) scaleResourceQuotaWithVMI(vmi *kubevirtv1.VirtualMachineInstance) error {
+	// Scale ResourceQuota through VMI resource specifications
+	rqs, err := h.rqCache.List(vmi.Namespace, labels.Everything())
+	if err != nil || len(rqs) == 0 {
+		return err
+	}
+
+	rqCpy := rqs[0].DeepCopy()
+	if _, ok := h.isVMIExistsMigrationVMsAnnotation(rqCpy, vmi.Name); ok {
+		return nil
+	}
+
+	needUpdate, toUpdate, rl := rqutils.CalculateScaleResourceQuotaWithVMI(rqCpy, vmi)
```

Comment:
nit: can we please rename `toUpdate` to something more meaningful like `rqToUpdate`?